mypyc/genops.py

```diff
@@ -1840,6 +1901,15 @@ def visit_name_expr(self, expr: NameExpr) -> Value:
             assert desc.result_type is not None
             return self.add(PrimitiveOp([], desc, expr.line))
 
+        if isinstance(expr.node, Var) and expr.node.is_final:
+            fvar = expr.node
+            if fvar.final_value is not None:  # this is safe even for non-native names
+                return self.load_final_literal_value(fvar.final_value, expr.line)
+            elif fullname.rpartition('.')[0] in self.modules:  # native final name
```

Comment:
There is similar logic in `is_native_ref_expr` that is probably worth extracting into its own helper and using.